### DISCUSSION TITLE
Difference between IDE0059 and CS0219

### DISCUSSION THREAD

**Discussion by The-Futurist** [dotnet/csharplang#7766] [General]
The first `var` assignment elicits IDE0059 and the second elicits CS0219, why?

```cs
namespace ConsoleApp3
{
    internal class Program
    {
        #region Static Methods
        static void Main(string[] args)
        {
            string brother = "John";
            string sister = "Jane";

            var message1 = $$$"""
                      {
                      "sister": "{{{sister}}}",
                      "brother": "{{{brother}}}",
                      }
                      """;

            var message2 = """
                      {
                      "sister": "Jane",
                      "brother": "John",
                      }
                      """;
        }
        #endregion
    }
}
```

**Comment by CyrusNajmabadi** [comment]
Both give IDE0059, which is a more extensive set of checks about a value being used.  

Only the latter gives CS0219 because that is a very old warning that that only ran on limited cases, and opts itself out in others.  In this case, it doesn't run on interpolated strings.  So you don't get any warnings there.

**Comment by CyrusNajmabadi** [comment]
Closing out as this is not a csharplang discussion.  Questions about warnings and analyzers can go to dotnet/roslyn.  Thanks!
